Core/include/Acts/Seeding/CandidatesForMiddleSp.hpp

```diff
@@ -0,0 +1,187 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2018-2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <tuple>
+#include <vector>
+
+namespace Acts {
+/// @brief A description of a triplet candidate.
+/// @tparam external_space_point_t  The external spacepoint type.
+template <typename external_space_point_t>
+struct TripletCandidate {
+  /// @brief Default Constructor
+  TripletCandidate() = default;
+
+  /// @brief constructor
+  /// @param b The bottom space point
+  /// @param m The middle space point
+  /// @param t The top space point
+  /// @param w The quality of the candidate
+  /// @param z The z coordinate of the origin
+  /// @param q Whether the candidate is high or low quality
+  TripletCandidate(external_space_point_t& b, external_space_point_t& m,
+                   external_space_point_t& t, float w, float z, bool q)
+      : bottom(&b), middle(&m), top(&t), weight(w), zOrigin(z), isQuality(q){};
+
+  external_space_point_t* bottom{nullptr};
+  external_space_point_t* middle{nullptr};
+  external_space_point_t* top{nullptr};
+  float weight{0.};
+  float zOrigin{0.};
+  bool isQuality{false};
+};
+
+/// @class CandidatesForMiddleSp
+/// The CandidatesForMiddleSp collects the triplet candidates given a
+/// fixed middle spacepoint. It internally stores the triplet candidates
+/// keeping only those with the higher quality.
+///
+/// @tparam external_space_point_t The external spacepoint type.
+
+template <typename external_space_point_t>
+class CandidatesForMiddleSp {
+ public:
+  using value_type = TripletCandidate<external_space_point_t>;
+
+  /// @brief constructor
+  CandidatesForMiddleSp() = default;
+  /// @brief Destructor
+  ~CandidatesForMiddleSp() = default;
+
+  /// @brief Setting maximum number of candidates to keep
+  /// @param n_low Maximum number of candidates in the low-quality collection
+  /// @param n_high Maximum number of candidates in the high-quality collection
+  void setMaxElements(std::size_t n_low, std::size_t n_high);
+
+  /// @brief Retrieve the triplet candidates, the resulting vector is already sorted,
+  /// elements with higher quality first
+  /// @returns Vector of triplet candidates
+  std::vector<value_type> storage();
```

Comment:
It is a bit strange that getter have a chance to modify the object. I.e. is not const method. I guess it has to be so in this case.